language/move-lang/src/cfgir/borrows/state.rs

```diff
@@ -668,13 +703,19 @@ impl BorrowState {
 
         let borrows = self.borrows.join(&other.borrows);
         let next_id = locals.len() + 1;
+        let acquired_resources = self.acquired_resources.clone();
+        let prev_errors = self.prev_errors;
         assert!(next_id == self.next_id);
         assert!(next_id == other.next_id);
+        assert!(acquired_resources == other.acquired_resources);
+        assert!(prev_errors == other.prev_errors);
```

Comment:
For a join do we need exact equality of the errors, or something like inclusion?

Reply:
It's exact. These are errors from the previous compiler pass for this function. If that changes between control flow paths. there is a problem